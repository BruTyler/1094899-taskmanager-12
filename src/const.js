export const Color = {
  BLACK: `black`,
  YELLOW: `yellow`,
  BLUE: `blue`,
  GREEN: `green`,
  PINK: `pink`,
};
