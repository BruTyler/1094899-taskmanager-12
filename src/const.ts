export enum Color {
  BLACK = `black`,
  YELLOW = `yellow`,
  BLUE = `blue`,
  GREEN = `green`,
  PINK = `pink`,
}

export const COLORS = Object.values(Color);

export enum RenderPosition {
  AFTERBEGIN = `afterbegin`,
  BEFOREEND = `beforeend`
}

export enum SortType {
  DEFAULT = `DEFAULT`,
  DATE_UP = `DATE up`,
  DATE_DOWN = `DATE down`
}

export enum RepeatingDaysMap {
  mo = 0b1,
  tu = 0b10,
  we = 0b100,
  th = 0b1000,
  fr = 0b10000,
  sa = 0b100000,
  su = 0b1000000
}

export enum UserAction {
  UPDATE_TASK = `UPDATE_TASK`,
  ADD_TASK = `ADD_TASK`,
  DELETE_TASK = `DELETE_TASK`
}

export enum UpdateType {
  PATCH = `PATCH`,
  MINOR = `MINOR`,
  MAJOR = `MAJOR`
}

export enum TaskMode {
  DEFAULT = `DEFAULT`,
  EDITING = `EDITING`
}

export enum FilterType {
  ALL = `all`,
  OVERDUE = `overdue`,
  TODAY = `today`,
  FAVORITES = `favorites`,
  REPEATING = `repeating`,
  ARCHIVE = `archive`
}

export const MenuItem = {
  ADD_NEW_TASK: `ADD_NEW_TASK`,
  TASKS: `TASKS`,
  STATISTICS: `STATISTICS`
};

export enum HTTPMethod {
  GET = `GET`,
  PUT = `PUT`,
  POST = `POST`,
  DELETE = `DELETE`,
}

export enum SuccessHTTPStatusRange {
  MIN = 200,
  MAX = 299,
}
