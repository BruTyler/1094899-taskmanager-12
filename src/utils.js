function generateRandomInt(min, max) {
  let rand = min - 0.5 + Math.random() * (max - min + 1);
  return Math.round(rand);
}

export function pickRandomElement(array) {
  if (array === undefined || array === null || array.length === 0) {
    throw new Error(`Incorrect array`);
  }
  const randomIndex = generateRandomInt(0, array.length - 1);
  return array[randomIndex];
}

export function pickRandomBool() {
  return generateRandomInt(0, 1) === 1;
}

export function pickRandomDate() {
  if (pickRandomBool() === true) {
    return null;
  }

  const currentDate = new Date();
  const weakOffset = 1000 * 60 * 60 * 24 * 7;
  const minTime = currentDate.getTime() - weakOffset;
  const maxTime = currentDate.getTime() + weakOffset;
  const randomTime = generateRandomInt(minTime, maxTime);
  return new Date(randomTime);
}

