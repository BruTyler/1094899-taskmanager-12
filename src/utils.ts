const generateRandomInt = (min, max) => {
  let rand = min - 0.5 + Math.random() * (max - min + 1);
  return Math.round(rand);
};

export const pickRandomElement = (array) => {
  if (array === undefined || array === null || array.length === 0) {
    throw new Error(`Incorrect array`);
  }
  const randomIndex = generateRandomInt(0, array.length - 1);
  return array[randomIndex];
};

export const pickRandomBool = () => {
  return Boolean(generateRandomInt(0, 1));
};

export const pickRandomDate = () => {
  if (pickRandomBool() === true) {
    return null;
  }

  const dayOffset = 7;
  const randomDayOffset = generateRandomInt(-dayOffset, +dayOffset);

  const currentDate = new Date();
  currentDate.setDate(currentDate.getDate() + randomDayOffset);

  return new Date(currentDate);
};

export const isTaskExpired = (dueDate) => {
  if (dueDate === null) {
    return false;
  }

  const currentDate = new Date();

  return currentDate.getTime() > dueDate.getTime();
};

export const isTaskExpiringToday = (dueDate) => {
  if (dueDate === null) {
    return false;
  }

  const currentDate = new Date();

  return currentDate.toDateString() === dueDate.toDateString();
};

export const isTaskRepeating = (repeating) => {
  return Object.values(repeating).includes(true);
};

export const humanizeTaskDueDate = (dueDate) => {
  return dueDate.toLocaleString(`en-US`, {day: `numeric`, month: `long`});
};

