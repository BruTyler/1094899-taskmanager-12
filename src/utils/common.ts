const generateRandomInt = (min: number, max: number): number => {
  const rand = min - 0.5 + Math.random() * (max - min + 1);
  return Math.round(rand);
};

export const pickRandomElement = <T extends any> (elements: Array<T>): T => {
  if (elements === undefined || elements === null || elements.length === 0) {
    throw new Error(`Incorrect array`);
  }
  const randomIndex = generateRandomInt(0, elements.length - 1);
  return elements[randomIndex];
};

export const pickRandomBool = (): boolean => {
  return Boolean(generateRandomInt(0, 1));
};

export const pickRandomDate = (): null | Date => {
  if (pickRandomBool() === true) {
    return null;
  }

  const dayOffset = 7;
  const randomDayOffset = generateRandomInt(-dayOffset, +dayOffset);

  const currentDate = new Date();
  currentDate.setDate(currentDate.getDate() + randomDayOffset);

  return new Date(currentDate);
};

export const transformToStringId = (word: string): string => {
  return word.replace(/\s/g, `-`).toLowerCase();
};
