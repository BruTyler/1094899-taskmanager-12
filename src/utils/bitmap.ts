import {RepeatingDays} from '../types';
import {RepeatingDaysMap} from '../const';


export const isTaskRepeating = (repeatingMask: number): boolean => {
  return repeatingMask > 0;
};

export const isDayRepeating = (repeatingMask: number, dayName: string): boolean => {
  return Boolean(repeatingMask & RepeatingDaysMap[dayName]);
};

export const getRepeatingDayNames = (): string[] => {
  return Object.keys(RepeatingDaysMap).filter((dayKey) => isNaN(Number(dayKey)));
};

export const reduceDaysToMask = (selectedDays: string[]): number => {
  return selectedDays.reduce(
      (accumulatedBit, dayKey) => accumulatedBit | RepeatingDaysMap[dayKey],
      0
  );
};

export const convertRepeatingToMask = (repeatingDays: RepeatingDays): number => {
  const repeatingDayNames = getRepeatingDayNames();

  const repeatingMask = repeatingDayNames
    .reduce(
        (accumulatedBit, dayKey) => {
          if (repeatingDays[dayKey]) {
            accumulatedBit += repeatingDays[dayKey] ? RepeatingDaysMap[dayKey] : 0;
          }

          return accumulatedBit;
        },
        0
    );

  return repeatingMask;
};

export const convertRepeatingToDays = (repeatingMask: number): Record<string, boolean> => {
  const repeatingDayNames = getRepeatingDayNames();

  const repeatingDays = {};
  repeatingDayNames.forEach((dayKey) => {
    repeatingDays[dayKey] = isDayRepeating(repeatingMask, dayKey);
  });

  return repeatingDays;
};
