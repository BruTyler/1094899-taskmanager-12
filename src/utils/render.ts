import {RenderPosition} from '../const';
import AbstractView from '../view/abstract';

export const render = (container: HTMLElement | AbstractView, child: ChildNode | AbstractView, place: RenderPosition): void => {
  if (container instanceof AbstractView) {
    container = container.getElement();
  }

  if (child instanceof AbstractView) {
    child = child.getElement();
  }

  switch (place) {
    case RenderPosition.AFTERBEGIN:
      container.prepend(child);
      break;
    case RenderPosition.BEFOREEND:
      container.append(child);
      break;
  }
};

export const renderTemplate = (container: HTMLElement | AbstractView, template: string, place: RenderPosition): void => {
  if (container instanceof AbstractView) {
    container = container.getElement();
  }

  container.insertAdjacentHTML(place, template);
};

export const createElement = (template: string): HTMLElement => {
  const newElement = document.createElement(`div`);
  newElement.innerHTML = template;

  return <HTMLElement> newElement.firstChild;
};

export const replace = (newChild: HTMLElement | AbstractView, oldChild: HTMLElement | AbstractView): void => {
  if (oldChild instanceof AbstractView) {
    oldChild = oldChild.getElement();
  }

  if (newChild instanceof AbstractView) {
    newChild = newChild.getElement();
  }

  const parent = oldChild.parentElement;

  if (parent === null || oldChild === null || newChild === null) {
    throw new Error(`Can't replace unexisting elements`);
  }

  parent.replaceChild(newChild, oldChild);
};

export const remove = (component: HTMLElement | AbstractView): void => {
  if (component === null) {
    return;
  }

  if (!(component instanceof AbstractView)) {
    throw new Error(`Can remove only components`);
  }

  component.getElement().remove();
  component.removeElement();
};
