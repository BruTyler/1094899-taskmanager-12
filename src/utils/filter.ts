import {FilterType} from '../const';
import {isTaskExpired, isTaskExpiringToday} from './task';
import {isTaskRepeating} from './bitmap';
import {Task} from '../types';

export const filter = {
  [FilterType.ALL]: (tasks: Task[]) => tasks.filter((task) => !task.isArchive),
  [FilterType.OVERDUE]: (tasks: Task[]) => tasks.filter((task) => isTaskExpired(task.dueDate)),
  [FilterType.TODAY]: (tasks: Task[]) => tasks.filter((task) => isTaskExpiringToday(task.dueDate)),
  [FilterType.FAVORITES]: (tasks: Task[]) => tasks.filter((task) => task.isFavorite),
  [FilterType.REPEATING]: (tasks: Task[]) => tasks.filter((task) => isTaskRepeating(task.repeatingMask)),
  [FilterType.ARCHIVE]: (tasks: Task[]) => tasks.filter((task) => task.isArchive)
};
