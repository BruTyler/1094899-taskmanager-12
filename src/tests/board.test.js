import BoardPresenter from '../presenter/board';
import {Color} from '../const';
import {createElement} from '../utils/render';

it(`Board-Presenter rendering`, () => {
  const task = {
    description: `some description`,
    dueDate: new Date(2020, 8, 4),
    repeatingDays: {
      mo: false,
      tu: false,
      we: false,
      th: false,
      fr: false,
      sa: false,
      su: false
    },
    color: Color.BLACK,
    isFavorite: true,
    isArchive: false,
  };

  const tasks = new Array(10).fill(task);
  const containerElement = createElement(`<div id="container-template"></div>`);

  const boardPresenter = new BoardPresenter(containerElement);
  boardPresenter.init(tasks);

  expect(containerElement).toMatchSnapshot();
});
