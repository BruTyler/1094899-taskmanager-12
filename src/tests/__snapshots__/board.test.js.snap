// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board-Presenter rendering 1`] = `
<div
  id="container-template"
>
  <section
    class="board container"
  >
    <div
      class="board__filter-list"
    >
      
    
      <a
        class="board__filter"
        data-sort-type="DEFAULT"
        href="#"
      >
        SORT BY DEFAULT
      </a>
      

      <a
        class="board__filter"
        data-sort-type="DATE up"
        href="#"
      >
        SORT BY DATE up
      </a>
      

      <a
        class="board__filter"
        data-sort-type="DATE down"
        href="#"
      >
        SORT BY DATE down
      </a>
      
  
    </div>
    <div
      class="board__tasks"
    >
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
      <article
        class="card card--black  "
      >
        
      
        <div
          class="card__form"
        >
          
        
          <div
            class="card__inner"
          >
            
          
            <div
              class="card__control"
            >
              
            
              <button
                class="card__btn card__btn--edit"
                type="button"
              >
                
              edit
            
              </button>
              
            
              <button
                class="card__btn card__btn--archive"
                type="button"
              >
                
              archive
            
              </button>
              
            
              <button
                class="card__btn card__btn--favorites card__btn--disabled"
                type="button"
              >
                
              favorites
            
              </button>
              
          
            </div>
            

          
            <div
              class="card__color-bar"
            >
              
            
              <svg
                class="card__color-bar-wave"
                height="10"
                width="100%"
              >
                
              
                <use
                  xlink:href="#wave"
                />
                
            
              </svg>
              
          
            </div>
            

          
            <div
              class="card__textarea-wrap"
            >
              
            
              <p
                class="card__text"
              >
                some description
              </p>
              
          
            </div>
            

          
            <div
              class="card__settings"
            >
              
            
              <div
                class="card__details"
              >
                
              
                <div
                  class="card__dates"
                >
                  
        
                  <div
                    class="card__date-deadline"
                  >
                    
          
                    <p
                      class="card__input-deadline-wrap"
                    >
                      
            
                      <span
                        class="card__date"
                      >
                        4 September
                      </span>
                      
          
                    </p>
                    
        
                  </div>
                  
      
                </div>
                
            
              </div>
              
          
            </div>
            
        
          </div>
          
      
        </div>
        
    
      </article>
    </div>
    <button
      class="load-more"
      type="button"
    >
      load more
    </button>
  </section>
</div>
`;
